internal/controller/serviceaccount_controller.go

```diff
@@ -171,17 +151,42 @@ func (r *serviceAccountReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *serviceAccountReconciler) shouldCreateOrRefreshImagePullSecret(
-	ctx context.Context, logger logr.Logger, sa *corev1.ServiceAccount,
-) (should bool, expiresAt time.Time, _ error) {
-	if !hasConfig(sa) {
-		logger.Info("ServiceAccount does not have configuration for image pull secret provisioning.")
-		return false, time.Time{}, nil
+func (r *serviceAccountReconciler) provisionSecretForPrincipal(
+	ctx context.Context, sa *corev1.ServiceAccount, principal string, principalIndex int,
+) (expiresAt time.Time, _ error) {
+	name := secretNameIndexed(sa, principalIndex)
+	logger := log.FromContext(ctx).WithValues("secret", name, "accountIndex", principalIndex)
```

Comment:
```suggestion
	logger := log.FromContext(ctx).WithValues("secret", name, "principalIndex", principalIndex)
```

Reply:
Or, we can log the principal, rather than its index

Reply:
I chose logging the principal.